src/util/records.py

```diff
@@ -1440,6 +1465,11 @@ def replace_record_references_batch(cr, id_mapping, model_src, model_dst=None, r
     for ir in indirect_references(cr, bound_only=True):
         if ir.table in ignores:
             continue
+        if ir.company_dependent_comodel:
+            if ir.company_dependent_comodel == model_src:
+                assert model_src == model_dst
+                raise Exception("Company dependent merge should be handled but don't support it for now")
```

Comment:
What's wrong with updating the values? Isn't it just replacing one id by another?
Something like:
```sql
                     UPDATE {table}
                        SET {column} = (
                                SELECT jsonb_object_agg(key, %s::jsonb->value)
                                  FROM jsonb_each_text({column})
                            )
                      WHERE {column} @? %s
```
with parameters `[Json(id_mapping), "$.* ? ({})".format(" || ".join(map("@ == {}".format, tuple(id_mapping)))})]`

Maybe you have some other constraints in mind, could you please elaborate?